src/main/java/edu/ksu/canvas/attendance/submitter/AssignmentValidator.java

```diff
@@ -54,32 +55,24 @@ public AttendanceAssignment validateCanvasAssignment(AttendanceAssignment assign
         Optional<Assignment> assignmentOptional = lookForAssignmentInCanvas(courseId,attendanceAssignment,canvasApiWrapperService,oauthToken);
 
         //Looks for discrepancy between the assignment in canvas and in database
-        if (anyDiscrepancy(assignmentOptional.get(), assignmentConfigurationFormSetup)){
+        if (!(assignmentOptional.get().getPointsPossible() - Double.parseDouble(assignmentConfigurationFormSetup.getAssignmentPoints()) < epsilon)) {
             LOG.debug("Discrepancy between Canvas and DB assignment. Point value of Canvas assignment is: " + assignmentOptional.get().getPointsPossible() +
                       " and point value of Database assignment is :" + attendanceAssignment.getAssignmentPoints());
             attendanceAssignment.setStatus(AttendanceAssignment.Status.CANVAS_AND_DB_DISCREPANCY);
         }
         return attendanceAssignment;
     }
 
-    public Boolean anyDiscrepancy(Assignment canvasAssignment, AttendanceAssignment dbAssignment){
-        if (canvasAssignment.getName() != dbAssignment.getAssignmentName()){
-            return false;
-        }
-        return canvasAssignment.getPointsPossible() == dbAssignment.getAssignmentPoints();
-
-    }
-
     /**
      * Returns true if configuration in the form and in the db is the same
      */
     private boolean isAssignmentConfigurationSaved(AttendanceAssignment assignmentConfigurationFromSetup, AttendanceAssignment attendanceAssignmentSaved) {
-        Boolean isSaved = assignmentConfigurationFromSetup.getAssignmentPoints().longValue() == attendanceAssignmentSaved.getAssignmentPoints().longValue();
-        isSaved = isSaved && (assignmentConfigurationFromSetup.getExcusedPoints().longValue() == attendanceAssignmentSaved.getExcusedPoints().longValue());
-        isSaved = isSaved && (assignmentConfigurationFromSetup.getAbsentPoints().longValue() == attendanceAssignmentSaved.getAbsentPoints().longValue());
-        isSaved = isSaved && (assignmentConfigurationFromSetup.getTardyPoints().longValue() == attendanceAssignmentSaved.getTardyPoints().longValue());
-        isSaved = isSaved && (assignmentConfigurationFromSetup.getPresentPoints().longValue() == attendanceAssignmentSaved.getPresentPoints().longValue());
-        return isSaved;
+
+        return assignmentConfigurationFromSetup.getAssignmentPoints().equals(attendanceAssignmentSaved.getAssignmentPoints())
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1067)